src/components/UrqlQuery.re

```diff
@@ -1,70 +1,57 @@
-[@bs.module "urql"] external queryComponent: ReasonReact.reactClass = "Query";
-
+[@bs.deriving abstract]
 type queryRenderPropsJs('a) = {
-  .
-  "fetching": bool,
-  "data": Js.Nullable.t('a),
-  "error": Js.Nullable.t(UrqlCombinedError.t),
-  "executeQuery":
-    option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult),
+  fetching: bool,
+  [@bs.optional] data: 'a,
+  [@bs.optional] error: UrqlCombinedError.t,
+  executeQuery: option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult)
 };
 
 type queryRenderProps('a) = {
   fetching: bool,
   data: option('a),
   error: option(UrqlCombinedError.t),
   executeQuery: option(Js.Json.t) => Js.Promise.t(UrqlTypes.operationResult),
-  response: UrqlTypes.response('a),
+  response: UrqlTypes.response('a)
 };
 
+module QueryJs = {
+  [@bs.module "urql"] [@react.component]
+  external make:
+    (~query: string, ~variables: Js.Json.t, ~requestPolicy: string, ~children: queryRenderPropsJs('a) => React.element) =>
+    React.element =
+    "Query";
+}
+
 let urqlDataToRecord = (result: queryRenderPropsJs('a)) => {
-  let data = result##data |> Js.Nullable.toOption;
-  let error = result##error |> Js.Nullable.toOption;
+  let data = result->dataGet;
+  let error = result->errorGet;
+  let fetching = result->fetchingGet;
 
   let response: UrqlTypes.response('a) =
-    switch (result##fetching, data, error) {
+    switch (fetching, data, error) {
     | (true, _, _) => Fetching
     | (false, Some(data), _) => Data(data)
     | (false, _, Some(error)) => Error(error)
     | (false, None, None) => NotFound
     };
 
   {
-    fetching: result##fetching,
+    fetching,
     data,
     error,
-    executeQuery: result##executeQuery,
+    executeQuery: result->executeQueryGet,
     response,
   };
 };
 
-[@bs.deriving abstract]
-type jsProps = {
-  query: string,
-  [@bs.optional]
-  variables: Js.Json.t,
-  requestPolicy: string,
+[@react.component]
+let make = (
+  ~query: string,
+  ~variables: Js.Json.t,
```

Comment:
In `urql`, `variables` is an optional prop. I'd like to see it typed as optional both in the `make` signature here and in the `QueryJs` binding, i.e.

```suggestion
  ~variables: option(Js.Json.t)=?
```

in both places. Does that make sense or am I missing something?